src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -383,6 +434,77 @@ public EventPipeEventMetaData(PinnedStreamReader reader, int length, int fileFor
         public int Level { get { return _eventRecord->EventHeader.Level; } }
 
         #region private 
+
+        /// <summary>
+        /// Reads the meta data for information specific to one event.  
+        /// </summary>
+        private void ReadEventMetaData(PinnedStreamReader reader, int fileFormatVersionNumber)
+        {
+            int eventId = (ushort)reader.ReadInt32();
+            _eventRecord->EventHeader.Id = (ushort)eventId;
+            Debug.Assert(_eventRecord->EventHeader.Id == eventId);  // No truncation
+
+            EventName = reader.ReadNullTerminatedUnicodeString();
+
+            // Deduce the opcode from the name.   
+            if (EventName.EndsWith("Start", StringComparison.OrdinalIgnoreCase))
```

Comment:
There has been a request to have EventPipe emit the opcode.  Do we want to just do this in CoreCLR instead of deducing it here?